### ISSUE TITLE
struct-tag linter false positive for empty options

### ISSUE THREAD

**Issue by wI2L** [mgechev/revive#208]
**Describe the bug**
When using the following struct tag `-,`, which is valid according the `json.Marshal` documentation, to encode a field with `-` as the name, the linter `struct-tag` reports the following warning: `unknown option '' in JSON tag`

**Expected behavior**
No warnings.

**Desktop (please complete the following information):**
 - linux/amd64
 - go1.12.5

**Additional context**
Example code:
```go
type x struct {
    H string `json:"-,"` // use "-" as key
}
```

I assume that a new value for the switch case located at https://github.com/mgechev/revive/blob/master/rule/struct-tag.go#L167 should be added for the empty string when the field name is `-`.

**Comment by chavacava** [comment]
@wI2L can you please check if the proposed fix works for you?
Merci.

**Comment by wI2L** [comment]
@chavacava Damn, you are quick !
I builded `revive` with your branch and I confirm that the warning is gone :tada: 

However, it catched a case for the following JSON tag that, while absolutely correct, looks badly explained to me.
```go
A string `json:","`
```
This JSON tag is invalid according to the **encoding/json** package because commas are not allowed in field names, while the message says: "option can not be empty in JSON tag". I think this message should be displayed only if a name is present before a comma, e.g `json:"a,"`. For the case I mention, another message of this style "invalid character in tag name".

>  The key name will be used if it's a non-empty string consisting of only Unicode letters, digits, and ASCII punctuation except quotation marks, backslash, and comma. 

https://golang.org/pkg/encoding/json/#Marshal
